src/main/java/org/jabref/gui/ai/components/aichat/AiChatComponent.java

```diff
@@ -1,317 +1,263 @@
-
-    // If the current content of the prompt is a history message, then this property is true.
-    // If user begins to edit or type a new text, then this property is false.
-    private final BooleanProperty showingHistoryMessage = new SimpleBooleanProperty(false);
+    private final AiChatLogic aiChatLogic;
 
-    @FXML private ScrollPane scrollPane;
-    @FXML private VBox chatVBox;
-    @FXML private HBox promptHBox;
-    @FXML private ExpandingTextArea userPromptTextArea;
-    @FXML private Button submitButton;
-    @FXML private StackPane stackPane;
+    @FXML private Loadable uiLoadableChatHistory;
+    @FXML private ChatHistoryComponent uiChatHistory;
+    @FXML private Button notificationsButton;
+    @FXML private ChatPromptComponent chatPrompt;
     @FXML private Label noticeText;
 
-    public AiChatComponent(AiPreferences aiPreferences, AiChatLogic aiChatLogic, String citationKey, DialogService dialogService, TaskExecutor taskExecutor) {
-        this.aiPreferences = aiPreferences;
-        this.aiChatLogic = aiChatLogic;
-        this.citationKey = citationKey;
+    public AiChatComponent(AiService aiService,
+                           String name,
+                           ObservableList<ChatMessage> chatHistory,
+                           ObservableList<BibEntry> entries,
+                           BibDatabaseContext bibDatabaseContext,
+                           DialogService dialogService,
+                           TaskExecutor taskExecutor
+    ) {
+        this.aiService = aiService;
+        this.entries = entries;
+        this.bibDatabaseContext = bibDatabaseContext;
         this.dialogService = dialogService;
         this.taskExecutor = taskExecutor;
 
+        this.aiChatLogic = new AiChatLogic(aiService, name, chatHistory, entries);
+
         ViewLoader.view(this)
                 .root(this)
                 .load();
     }
 
     @FXML
     public void initialize() {
-        userPromptTextArea.setOnKeyPressed(keyEvent -> {
-            if (keyEvent.getCode() == KeyCode.DOWN) {
-                // Do not go down in the history.
-                if (currentUserMessageScroll.get() != NEW_NON_EXISTENT_MESSAGE) {
-                    showingHistoryMessage.set(true);
-                    currentUserMessageScroll.set(currentUserMessageScroll.get() - 1);
-
-                    // There could be two effects after setting the properties:
-                    // 1) User scrolls to a recent message, then we should properly update the prompt text.
-                    // 2) Scroll is set to -1 (which is NEW_NON_EXISTENT_MESSAGE) and we should clear the prompt text.
-                    // On the second event currentUserMessageScroll will be set to -1 and showingHistoryMessage
-                    // will be true (this is important).
-                }
-            } else if (keyEvent.getCode() == KeyCode.UP) {
```

Comment:
Thus functionality is removed. Please re-add it.

Do we need a list of requirements of the chat tab? This removal is an indicator that we do.

Maybe, it's time to play around [OpenFastTrace](https://github.com/itsallcode/openfasttrace)  -- user guide: https://github.com/itsallcode/openfasttrace/blob/main/doc/user_guide.md

Reply:
no, it's there

https://github.com/InAnYan/jabref/blob/e68ccabf4d35f7fd073cbefdf8242452a1399167/src/main/java/org/jabref/gui/ai/components/aichat/chatprompt/ChatPromptComponent.java#L78

it just didn't worked, I'll push the commit right now, where it works